### PR TITLE
Shard pr

### PR DESCRIPTION
**Pull request by minhtriet**
**Description**
This PR addresses an issue with the way gradients were being calculated during the `pmap` parallelization in our model training. Specifically, the gradient computation in multi-device setups was not being aggregated, leading to bigger loss than expected

**Changes Made:**
- Fixed the gradient calculation logic within the pmap framework to ensure that gradients are correctly averaged across devices.
- Reformat the code and make some changes to the code so that the gradients is calculated more reasonably.

**Baseline Without pmap**
`{"Date": "09/24/2025, 20:30:54", "Num qubits": 6, "Test loss": 1.7924214053224077, "Num gates": 8, "Full measurements": "prob", "Group members": ["_90_deg_x_rot", "_180_deg_x_rot", "_270_deg_x_rot"], "Epochs": 5, "Train losses": [7.19446756372112, 5.758194641059593, 4.59227077508878, 3.507013883804911, 2.9204264198774372], "Test losses": [{"2": 3.0524761015084767}], "LR": 0.01, "Batch size": 10}`

**Before (`Sharding` branch)**
`{"Date": "09/24/2025, 20:00:05", "Num qubits": 6, "Test loss": 5.083189883078406, "Num gates": 8, "Full measurements": "prob", "Group members": ["_90_deg_x_rot", "_180_deg_x_rot", "_270_deg_x_rot"], "Epochs": 5, "Train losses": [11.465957680732707, 8.328580011729274, 5.795973478171395, 4.063220143942548, 2.8973570868150564], "Test losses": [{"2": 5.394668879535257}], "LR": 0.01, "Batch size": 10}`

**After (This branch)**
`{"Date": "09/28/2025, 16:32:29", "Num qubits": 6, "Test loss": 4.104882570547979, "Num gates": 8, "Full measurements": "prob", "Group members": ["_90_deg_x_rot", "_180_deg_x_rot", "_270_deg_x_rot"], "Epochs": 5, "Train losses": [9.741145172909848, 1.2031453173599436, 0.36010744360332547, 0.4449008122350716, 0.5348045051594273], "Test losses": [{"2": 4.435546901306365}], "LR": 0.04, "Batch size": 10}`

### CODE REVIEW

File: graddft_qnn/helper/pmap_training.py [hunk 1]
```diff
@@ -32,8 +30,10 @@ def per_device(p, mol, y):
     # run pmap
     loss_value, grads = loss_grad_pmap(parameters, mol_jax, targets)
 
+    grads_mean = jax.tree_map(lambda x: x / len(batch["coordinates"]), grads)
```
Comment (minhtriet) [inline graddft_qnn/helper/pmap_training.py:33]: So I average here
